src/ts/components/fullscreentogglebutton.ts

```diff
@@ -20,28 +20,32 @@ export class FullscreenToggleButton extends ToggleButton<ToggleButtonConfig> {
   configure(player: PlayerAPI, uimanager: UIInstanceManager): void {
     super.configure(player, uimanager);
 
-    let fullscreenStateHandler = () => {
-      if (player.getViewMode() === player.exports.ViewMode.Fullscreen) {
-        this.on();
-      } else {
-        this.off();
-      }
+    let isFullScreenAvailable = () => {
+      return player.isViewModeAvailable(player.exports.ViewMode.Fullscreen);
     };
 
-    let fullscreenAvailableHandler = () => {
-      if (player.isViewModeAvailable(player.exports.ViewMode.Fullscreen)) {
-        this.show();
-      } else {
-        this.hide();
-      }
+    let fullscreenStateHandler = () => {
+      player.getViewMode() === player.exports.ViewMode.Fullscreen ? this.on() : this.off();
     };
 
-    uimanager.getConfig().events.onUpdated.subscribe(fullscreenAvailableHandler);
+    let fullscreenAvailabilityChangedHandler = () => {
+      isFullScreenAvailable() ? this.show() : this.hide();
+    };
 
     player.on(player.exports.PlayerEvent.ViewModeChanged, fullscreenStateHandler);
 
+    // Available only in our native SDKs for now
+    if ((player.exports.PlayerEvent as any).ViewModeAvailabilityChanged) {
+      player.on(
+        (player.exports.PlayerEvent as any).ViewModeAvailabilityChanged,
+        fullscreenAvailabilityChangedHandler,
```

Comment:
You could actually check for `event.available` here, right?

Reply:
Yes I could do that some reasons I decided against doing it:
- we have this callback already and call it also initially
- we are doing it like that on many other places as well
- we don't have the type of the event yet. So we would either need to declare the type somewhere or use `any`

Reply:
Hm, k. Fine 4 me as well. Point 2 is not a valid reason imo 😉